condor_git_config.py

```diff
@@ -91,127 +98,174 @@ class ConfigCache(object):
                 timeout=30,
                 universal_newlines=True,
             )
+            self._config_meta["timestamp"] = time.time()
         else:
-            subprocess.check_output(
-                ['git', 'pull'],
-                timeout=30,
-                cwd=repo_path,
-                universal_newlines=True
-            )
-        self._update_metadata()
+            try:
+                subprocess.check_output(
+                    ["git", "pull"], timeout=30, cwd=repo_path, universal_newlines=True
+                )
+            except (subprocess.CalledProcessError, subprocess.TimeoutExpired):
+                pass
+            else:
+                self._config_meta["timestamp"] = time.time()
 
 
 class ConfigSelector(object):
-    def __init__(self, pattern, blacklist, whitelist, recurse):
+    """
+    Selector for a configuration file iterator
+    """
+
+    def __init__(self, pattern, blacklist, whitelist, recurse: bool):
         self.pattern = self._prepare_re(pattern)
-        self.blacklist = self._prepare_re(blacklist, '(?!)')
+        self.blacklist = self._prepare_re(blacklist, default="(?!)")
         self.whitelist = self._prepare_re(whitelist)
         self.recurse = recurse
 
     @staticmethod
-    def _prepare_re(pieces, default='.*'):
+    def _prepare_re(pieces, default=".*") -> "re.Pattern":
         if not pieces:
             return re.compile(default)
         if len(pieces) == 1:
             return re.compile(pieces[0])
         else:
-            return re.compile('|'.join('(?:%s)' % piece for piece in pieces))
+            return re.compile("|".join("(?:%s)" % piece for piece in pieces))
 
-    def get_paths(self, config_cache):
+    def get_paths(self, config_cache: ConfigCache) -> Iterable[Path]:
+        pattern, blacklist, whitelist = self.pattern, self.blacklist, self.whitelist
         for rel_path in config_cache:
-            if not self.recurse and os.path.dirname(rel_path):
+            if not self.recurse and rel_path.parent != Path("."):
                 continue
-            if self.pattern.search(rel_path):
-                if not self.blacklist.search(rel_path) or self.whitelist.search(rel_path):
+            str_path = str(rel_path)
+            if pattern.search(str_path):
+                if not blacklist.search(str_path) or whitelist.search(str_path):
                     yield config_cache.repo_path(rel_path)
 
 
-def include_configs(path_key, config_cache, config_selector, destination=sys.stdout):
+def include_configs(
+    path_key: str,
+    config_cache: ConfigCache,
+    config_selector: ConfigSelector,
+    destination=sys.stdout,
+):
     with config_cache:
-        config_cache.refresh()
-        print('%s = %s' % (path_key, config_cache.repo_path()), file=destination)
+        print("#", json.dumps(config_cache.stats))
+        print("%s = %s" % (path_key, config_cache.repo_path()), file=destination)
```

Comment:
I guess those prints are intended to go to destination specified above specifically?
Otherwise you might consider also using the logger.

Reply:
These are all writing to the proper stream now.